spring-cloud-starter-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/single/security/DataFlowClientAutoConfigurationAgaintstServerTests.java

```diff
@@ -104,6 +105,30 @@ public void usingUserWithViewRoles() throws Exception {
 		context.close();
 	}
 
+	@Test
+	public void usingUserWithViewRolesWithOauth() throws Exception {
+
+		final AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+		TestPropertyValues.of(
+				"spring.cloud.dataflow.client.server-uri=" + "http://localhost:"
+						+ localDataflowResource.getDataflowPort(),
+				"spring.cloud.dataflow.client.authentication.client-id=myclient",
+				"spring.cloud.dataflow.client.authentication.client-secret=mysecret",
+				"spring.cloud.dataflow.client.authentication.token-uri=" + "http://localhost:"
+						+ oAuth2ServerResource.getOauth2ServerPort() + "/oauth/token",
+				"spring.cloud.dataflow.client.authentication.scope=dataflow.view").applyTo(context);
+		context.register(TestApplication2.class);
+		context.refresh();
+
+		final DataFlowOperations dataFlowOperations = context.getBean(DataFlowOperations.class);
+		final AboutResource about = dataFlowOperations.aboutOperation().get();
+
+		assertNotNull(about);
+		assertEquals("myclient", about.getSecurityInfo().getUsername());
+		assertEquals(1, about.getSecurityInfo().getRoles().size());
+		context.close();
```

Comment:
Nitpicky on the test.   We should move the close context to an @After to guarantee that the context is closed.    (But that can be a different PR).